device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/auth/SignatureHelper.java

```diff
@@ -80,13 +80,9 @@
             hMacSha256.init(secretKey);
             encryptedSig = hMacSha256.doFinal(sig);
         }
-        catch (NoSuchAlgorithmException e)
+        catch (NoSuchAlgorithmException | InvalidKeyException e)
         {
-            // should never happen, since the algorithm is hard-coded.
-        }
-        catch (InvalidKeyException e)
-        {
-            // should never happen, since the input key type is hard-coded.
+            // should never happen, since the algorithm and input key are hard-coded.
```

Comment:
might be worth adding a comment stating that the only reason we catch and swallow this is to avoid having to declare these exceptions in the method signature.
also, should it say "will never happen", definitively